ambry-api/src/main/java/com.github.ambry/clustermap/ClusterParticipant.java

```diff
@@ -74,6 +74,16 @@ void registerPartitionStateChangeListener(StateModelListenerType listenerType,
    */
   ReplicaSyncUpManager getReplicaSyncUpManager();
 
+  /**
+   * Update disk/replica infos associated with current data node in cluster (this occurs when replica addition/removal
+   * on current node is complete and local changes will be broadcast to all listeners in this cluster)
+   * @param replicaId the {@link ReplicaId} whose info should be updated on current node
+   * @param shouldPresent Whether the replica info should be present. When {@code true}, replica info will be added if
```

Comment:
Minor: Feel `addOrRemove` is better.

Reply:
`addOrRemove` seems to require an enum. We cannot tell if it should be added or removed by looking at the boolean value (true/false).  Aside from `shouldPresent`, I also consider `shouldExist`.  I am still open to other proper names.

Reply:
`shouldExist` feels better.

Reply:
ok, will make the change.

Reply:
@ankagrawal  I guess you are suggesting `addReplicaToDataNodeInfo` and `removeReplicaFromDataNodeInfo`.  Yes, we could but a single method with boolean variable would suffice and seems to be more concise. In addition, updating data node info only applies to `HelixParticipant`. That is, we don't need to introduce one more unnecessary method to other implementation of `ClusterParticipant` (i.e. StaticClusterParticipant).